nav2_costmap_2d/include/nav2_costmap_2d/costmap_2d_ros.hpp

```diff
@@ -134,6 +134,27 @@ class Costmap2DROS : public nav2_util::LifecycleNode
    */
   nav2_util::CallbackReturn on_shutdown(const rclcpp_lifecycle::State & state) override;
 
+  /**
+   * @brief override on_rcl_preshutdown() as empty
+   * [reason] costmap only could be created by its parents like planner/controller_server
+   * [reason] costmap should react to ctrl+C later than its parents to avoid nullptr-accessed
+   * so the reaction must be later than its parent
+   *
+   * thus, it's a more perfect way to make on_rcl_preshutdown() as empty function
+   * only joint in planner/controller_server->on_deactivate()
+   * and planner/controller_server->on_cleanup()
+   *
+   * !!! a mention !!!
+   * if any other place use this class (Costmap2DROS)
+   * it's neccessary to let costmap->deactivate() joint in parent->deactivate()
+   * and let costmap->cleanup() joint in parent->cleanup()
+   */
+  void on_rcl_preshutdown() override
```

Comment:
This is not a viable solution, destroying bond and shutting things down is absolutely necessary.